include/arch/arc/v2/linker.ld

```diff
@@ -26,6 +26,12 @@
 		#define RAMABLE_REGION SRAM
 	#endif
 #endif
+/* Non cacheable memory region, can be overridden by SOC */
```

Comment:
@tejlmand is this a job for snippets?

Reply:
The `nocache.ld` is already being included as a snippet.
https://github.com/zephyrproject-rtos/zephyr/blob/2be4a537afa19878533909cea95d3fb96f6ffa98/arch/common/CMakeLists.txt#L41-L44

so instead of adding those `#ifdef` to all linker.ld files, then we could adjust the snippet instead, like https://github.com/zephyrproject-rtos/zephyr/blob/main/arch/common/nocache.ld:
```
#ifdef CONFIG_ARCH_HAS_NOCACHE_MEMORY_SUPPORT
} GROUP_DATA_LINK_IN(NOCACHE_REGION, NOCACHE_REGION)
#else
} GROUP_DATA_LINK_IN(RAMABLE_REGION, RAMABLE_REGION)
#endif
```

One concern I have with this, is the fact that this snippet is still injected in the ram section here:
https://github.com/zephyrproject-rtos/zephyr/blob/2be4a537afa19878533909cea95d3fb96f6ffa98/include/arch/arm/aarch32/cortex_m/scripts/linker.ld#L251-L254

So we should consider adding an extra nocache snippet here:
https://github.com/zephyrproject-rtos/zephyr/blob/2be4a537afa19878533909cea95d3fb96f6ffa98/cmake/extensions.cmake#L1170-L1179

that are then used when `CONFIG_ARCH_HAS_NOCACHE_MEMORY_SUPPORT=y`.

Reply:
Good time for a bit of promotion to https://github.com/zephyrproject-rtos/zephyr/issues/40146

Reply:
> so instead of adding those #ifdef to all linker.ld files, then we could adjust the snippet instead, like 
> ```
> #ifdef CONFIG_ARCH_HAS_NOCACHE_MEMORY_SUPPORT
> } GROUP_DATA_LINK_IN(NOCACHE_REGION, NOCACHE_REGION)
> #else
> } GROUP_DATA_LINK_IN(RAMABLE_REGION, RAMABLE_REGION)
> #endif
> ```

@tejlmand Just to make sure I'm understanding this snippet correctly, in the case of the Cortex M4 would the `NOCACHE_REGION` be defined by the SOC linker file? In that case, I think that a new KConfig symbol needs to be defined, as SOCs with a Cortex M7 also select `CONFIG_ARCH_HAS_NOCACHE_MEMORY_SUPPORT`, but do not define a `NOCACHE_REGION`, correct?

Reply:
> @tejlmand Just to make sure I'm understanding this snippet correctly, in the case of the Cortex M4 would the NOCACHE_REGION be defined by the SOC linker file? 

Something along that was my though.
The soc linker file or the general arch linker script, where it could be depending on the devicetree information.
Similar to the SRAMx regions, and the description of #40146.

> In that case, I think that a new KConfig symbol needs to be defined, as SOCs with a Cortex M7 also select CONFIG_ARCH_HAS_NOCACHE_MEMORY_SUPPORT, but do not define a NOCACHE_REGION, correct?

Good point.
So some thinking must be done if going into this direction.